src/m365/spfx/commands/project/project-upgrade/upgrade-1.22.0.ts

```diff
@@ -67,43 +61,66 @@ import { FN026001_CFG_SASS_schema } from './rules/FN026001_CFG_SASS_schema.js';
-  new FN001028_DEP_microsoft_sp_list_subscription('1.22.0-rc.0'),
-  new FN001029_DEP_microsoft_sp_loader('1.22.0-rc.0'),
-  new FN001030_DEP_microsoft_sp_module_interfaces('1.22.0-rc.0'),
-  new FN001031_DEP_microsoft_sp_odata_types('1.22.0-rc.0'),
-  new FN001032_DEP_microsoft_sp_page_context('1.22.0-rc.0'),
-  new FN001013_DEP_microsoft_decorators('1.22.0-rc.0'),
-  new FN001034_DEP_microsoft_sp_adaptive_card_extension_base('1.22.0-rc.0'),
-  new FN002001_DEVDEP_microsoft_sp_build_web('1.22.0-rc.0', false),
-  new FN002002_DEVDEP_microsoft_sp_module_interfaces('1.22.0-rc.0'),
+  new FN001001_DEP_microsoft_sp_core_library('1.22.0'),
+  new FN001002_DEP_microsoft_sp_lodash_subset('1.22.0'),
+  new FN001003_DEP_microsoft_sp_office_ui_fabric_core('1.22.0'),
+  new FN001004_DEP_microsoft_sp_webpart_base('1.22.0'),
+  new FN001011_DEP_microsoft_sp_dialog('1.22.0'),
+  new FN001012_DEP_microsoft_sp_application_base('1.22.0'),
+  new FN001014_DEP_microsoft_sp_listview_extensibility('1.22.0'),
+  new FN001021_DEP_microsoft_sp_property_pane('1.22.0'),
+  new FN001023_DEP_microsoft_sp_component_base('1.22.0'),
+  new FN001024_DEP_microsoft_sp_diagnostics('1.22.0'),
+  new FN001025_DEP_microsoft_sp_dynamic_data('1.22.0'),
+  new FN001026_DEP_microsoft_sp_extension_base('1.22.0'),
+  new FN001027_DEP_microsoft_sp_http('1.22.0'),
+  new FN001028_DEP_microsoft_sp_list_subscription('1.22.0'),
+  new FN001029_DEP_microsoft_sp_loader('1.22.0'),
+  new FN001030_DEP_microsoft_sp_module_interfaces('1.22.0'),
+  new FN001031_DEP_microsoft_sp_odata_types('1.22.0'),
+  new FN001032_DEP_microsoft_sp_page_context('1.22.0'),
+  new FN001013_DEP_microsoft_decorators('1.22.0'),
+  new FN001034_DEP_microsoft_sp_adaptive_card_extension_base('1.22.0'),
+  new FN002001_DEVDEP_microsoft_sp_build_web('1.22.0', false),
+  new FN002002_DEVDEP_microsoft_sp_module_interfaces('1.22.0'),
   new FN002004_DEVDEP_gulp('4.0.2', false),
   new FN002007_DEVDEP_ajv('6.12.6', false),
   new FN002021_DEVDEP_rushstack_eslint_config('4.5.2'),
-  new FN002022_DEVDEP_microsoft_eslint_plugin_spfx('1.22.0-rc.0'),
-  new FN002023_DEVDEP_microsoft_eslint_config_spfx('1.22.0-rc.0'),
+  new FN002022_DEVDEP_microsoft_eslint_plugin_spfx('1.22.0'),
+  new FN002023_DEVDEP_microsoft_eslint_config_spfx('1.22.0'),
   new FN002026_DEVDEP_typescript('~5.8.0'),
   new FN002029_DEVDEP_microsoft_rush_stack_compiler_5_3('0.1.0', false),
-  new FN002030_DEVDEP_microsoft_spfx_web_build_rig('1.22.0-rc.0'),
+  new FN002030_DEVDEP_microsoft_spfx_web_build_rig('1.22.0'),
   new FN002031_DEVDEP_rushstack_heft('1.1.2'),
   new FN002032_DEVDEP_typescript_eslint_parser('8.46.2'),
   new FN002033_DEVDEP_css_loader('7.1.2'),
-  new FN002034_DEVDEP_microsoft_spfx_heft_plugins('1.22.0-rc.0'),
+  new FN002034_DEVDEP_microsoft_spfx_heft_plugins('1.22.0'),
   new FN002035_DEVDEP_types_heft_jest('1.0.2'),
-  new FN010001_YORC_version('1.22.0-rc.0'),
+  new FN010001_YORC_version('1.22.0'),
   new FN010011_YORC_useGulp(false),
```

Comment:
I know this was not part of this change but when performing the upgrade from 1.21.1 to 1.22.0 I noticed this rule states that 

<img width="454" height="208" alt="Image" src="https://github.com/user-attachments/assets/9157e335-20e8-4915-9a15-a74e3d0592cb" />

In general it is ok and something verly small but technically previously (so before SPFx 1.22) such property did not exist in the yo-rc so it would be a bit better to state `Add useGulp property in .yo-rc.json` instead of `update`

https://github.com/pnp/cli-microsoft365/blob/c16a3a7f7af4bbe91a102af0ea2726722897020e/src/m365/spfx/commands/project/project-upgrade/rules/FN010011_YORC_useGulp.ts#L19C13-L19C51